### PR TITLE
Proposal: filebeat line publishing improvements

### PR DESCRIPTION
**Pull request by urso**
some refactoring in event output for use with filebeat.

- Add support for async, non-blocking (guaranteed) publishing and registrar updates

### CODE REVIEW

File: filebeat/beat/publish.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		out:    out,
+		client: client,
+	}
+}
+
+func (pub *logPublisher) start() {
+	go func() {
+		logp.Info("Start sending events to output")
+
+		ticker := time.NewTicker(5 * time.Second)
```
Comment (ruflin) [inline filebeat/beat/publish.go]: This time should probably be based on a constant or even configurable.

File: libbeat/outputs/logstash/client.go [hunk 2]
```diff
@@ -60,15 +64,29 @@ var (
 
 func newLumberjackClient(
 	conn TransportClient,
+	compressLevel int,
```
Comment (ruflin) [inline libbeat/outputs/logstash/client.go]: Could we introduce the compressLevel change in a separate PR?

File: libbeat/outputs/logstash/client.go [hunk 3]
```diff
@@ -263,8 +292,8 @@ func (l *lumberjackClient) sendCompressed(payload []byte) error {
 		return err
 	}
 
-	_, err := l.Write(payload)
-	return err
+	l.Write(payload)
```
Comment (ruflin) [inline libbeat/outputs/logstash/client.go]: What happens to the error returned by Write? Not needed anymore?

**Comment by ruflin** [comment]
I would suggest to split up this PR in 2 or 3 PR's if possible:
- Introduction of Compression Level. Passing of options.
- Renaming of confirm to guaranteed option
- Async load balancer

The first two I think are simple enhancements without lots of discussions needed, the third one we should probably look at again separately. Having it as a separated PR will make the change smaller and discussion easier. Currently the changes to the config file are missing (just as a note).

**Comment by urso** [comment]
all but async output mode (pipelining) ported to master. async output mode will be ported to master if required, but not for now, as not really required.

**Pull request closed without merging**